examples/platforms/efr32mg12/alarm.c

```diff
@@ -190,27 +202,33 @@ void efr32AlarmProcess(otInstance *aInstance)
             new_expires_microsec = (now + dt) * US_IN_MS;
             status = RAIL_SetTimer(gRailHandle, new_expires_microsec, RAIL_TIME_ABSOLUTE, RAILCb_TimerExpired);
 
-            if (status == RAIL_STATUS_NO_ERROR)
-            {
-                sIsRunning = true;
-            }
-            else
+            if (status != RAIL_STATUS_NO_ERROR)
             {
-                sIsRunning = false;
-                otLogCritPlat("Alarm set timer, status: %d, expires: %u, dt: %u, now: %u", status, sAlarmT0 + sAlarmDt,
-                              dt, now);
+                new_expires_microsec = dt * US_IN_MS;
+                status = RAIL_SetTimer(gRailHandle, new_expires_microsec, RAIL_TIME_DELAY, RAILCb_TimerExpired);
+
+                if (status != RAIL_STATUS_NO_ERROR)
+                {
+                    now = otPlatAlarmMilliGetNow();
```

Comment:
nit: can we put the call to `otPlatAlarmMilliGetNow()` in the argument list for `otLogCritPlat()`?

Reply:
Added the recommended change.